src/core/qgsogrutils.cpp

```diff
@@ -236,6 +244,16 @@ QVariant QgsOgrUtils::getOgrFeatureAttribute( OGRFeatureH ogrFet, const QgsField
         break;
       }
 
+      case QVariant::Map:
+      {
+        //it has to be JSON
+        //it's null if no json format
+        if ( encoding )
+          value = QJsonDocument::fromJson( encoding->toUnicode( OGR_F_GetFieldAsString( ogrFet, attIndex ) ).toUtf8() ).toVariant();
```

Comment:
I think the extra UTF8 conversion is not required

```suggestion
           value = QJsonDocument::fromJson( encoding->toUnicode( OGR_F_GetFieldAsString( ogrFet, attIndex ) ) ).toVariant();
```

Reply:
I did that to have a proper `QByteArray` in the end for `fromJson`.

Reply:
Good point. Let's stick to that.